exonum/src/storage/proof_list_index/proof.rs

```diff
@@ -79,12 +84,28 @@ impl<V: StorageValue> ListProof<V> {
     ///
     /// If the proof is valid, a vector with indices and references to elements is returned.
     /// Otherwise, `Err` is returned.
-    pub fn validate(&self, merkle_root: Hash, len: u64) -> Result<Vec<(u64, &V)>, ListProofError> {
+    pub fn validate(
+        &self,
+        expected_list_hash: Hash,
+        len: u64,
```

Comment:
Forgot to mention in the review: I see on compelling reasons to provide the list length from the external source. Where are you supposed to get it from in most cases? It could make more sense to supply the length with the proof.